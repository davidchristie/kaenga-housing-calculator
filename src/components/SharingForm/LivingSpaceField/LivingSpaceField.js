import React, { Component, PropTypes } from 'react'
import { Checkbox } from 'react-bootstrap'

class LivingSpaceField extends Component {
  render () {
    const { input: { onChange, value } } = this.props
    return (
      <Checkbox
        onChange={onChange}
        value={value}
      >
        Living space (including kitchens)
      </Checkbox>
    )
  }
}

LivingSpaceField.defaultProps = {
  input: {}
}

LivingSpaceField.propTypes = {
  input: PropTypes.shape({
    onChange: PropTypes.func,
    value: PropTypes.any
  }).isRequired
}

export default LivingSpaceField
