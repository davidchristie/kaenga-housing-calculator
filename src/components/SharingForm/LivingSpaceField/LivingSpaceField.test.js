import React from 'react'
import ReactTestUtils from 'react-addons-test-utils'

import LivingSpaceField from './LivingSpaceField'

const shallowRenderer = ReactTestUtils.createRenderer()

it('matches snapshot', () => {
  expect(shallowRenderer.render(<LivingSpaceField />)).toMatchSnapshot()
})
