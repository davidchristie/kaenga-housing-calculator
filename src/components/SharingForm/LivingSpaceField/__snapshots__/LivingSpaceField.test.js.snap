exports[`test matches snapshot 1`] = `
<Checkbox
  bsClass="checkbox"
  disabled={false}
  inline={false}
  onChange={undefined}
  value={undefined}>
  Living space (including kitchens)
</Checkbox>
`;
